File: sayDigits.py

alphaDigits = ["zero","one","two","three","four","five","six","seven","eight","nine"]

def sayDigits(n :int):
    if (n==0):
        return
    sayDigits(int(n/10))
    rem = int(n%10)
    print(alphaDigits[rem],end=" ")

if __name__=='__main__':
    sayDigits(129934)
    print()

File: sayDigits.cpp
#include <iostream>
#include <vector>
using namespace std;
vector<string> alphaDigits = {"zero","one","two","three","four","five","six","seven","eight","nine"};

void sayDigit(int num){
    if (num==0) return;

    sayDigit(num/10);
    int rem = num%10;
    cout<<alphaDigits[rem]<<" ";
}
int main()
{
    sayDigit(123);
    return 0;
}